python/ray/data/_internal/logical/operators/all_to_all_operator.py

```diff
@@ -0,0 +1,92 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from ray.data._internal.block_list import BlockList
+from ray.data._internal.compute import BlockTransform
+from ray.data._internal.execution.interfaces import PhysicalOperator, RefBundle
+from ray.data._internal.execution.operators.all_to_all_operator import AllToAllOperator
+from ray.data._internal.logical.interfaces import LogicalOperator
+
+from ray.data._internal.stats import StatsDict
+
+
+class AbstractAllToAll(LogicalOperator):
+    """Abstract class for logical operators should be converted to physical
+    AllToAllOperator.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        input_op: LogicalOperator,
+        fn: Callable[[BlockList, bool, Callable, dict], Tuple[BlockList, dict]],
```

Comment:
We shouldn't reference the legacy BlockList abstractions in the new packages.

Reply:
@ericl - removed.